accelerator/multi_region_modules/hubNetworking.bicep

```diff
@@ -0,0 +1,1164 @@
+]
+
+@sys.description('Set Parameter to false to skip the addition of a Private DNS Zone for Azure Backup.')
+param parPrivateDnsZoneAutoMergeAzureBackupZone bool = true
+
+@sys.description('Resource ID of Failover VNet for Private DNS Zone VNet Failover Links')
+param parVirtualNetworkIdToLinkFailover string = ''
+
+@sys.description('''Resource Lock Configuration for Private DNS Zone(s).
+
+- `kind` - The lock settings of the service which can be CanNotDelete, ReadOnly, or None.
+- `notes` - Notes about this lock.
+
+''')
+param parPrivateDNSZonesLock lockType = {
+  kind: 'None'
+  notes: 'This lock was created by the ALZ Bicep Hub Networking Module.'
+}
+
+@sys.description('Switch to enable/disable VPN virtual network gateway deployment.')
+param parVpnGatewayEnabled bool = true
+
+//ASN must be 65515 if deploying VPN & ER for co-existence to work: https://docs.microsoft.com/en-us/azure/expressroute/expressroute-howto-coexist-resource-manager#limits-and-limitations
+@sys.description('Configuration for VPN virtual network gateway to be deployed.')
+param parVpnGatewayConfig object = {
+  name: '${parCompanyPrefix}-Vpn-Gateway'
+  gatewayType: 'Vpn'
+  sku: 'VpnGw1'
+  vpnType: 'RouteBased'
+  generation: 'Generation1'
+  enableBgp: false
+  activeActive: false
+  enableBgpRouteTranslationForNat: false
+  enableDnsForwarding: false
+  bgpPeeringAddress: ''
+  bgpsettings: {
+    asn: 65515
+    bgpPeeringAddress: ''
+    peerWeight: 5
+  }
+  vpnClientConfiguration: {}
+}
+
+@sys.description('Switch to enable/disable ExpressRoute virtual network gateway deployment.')
+param parExpressRouteGatewayEnabled bool = true
+
+@sys.description('Configuration for ExpressRoute virtual network gateway to be deployed.')
+param parExpressRouteGatewayConfig object = {
+  name: '${parCompanyPrefix}-ExpressRoute-Gateway'
+  gatewayType: 'ExpressRoute'
+  sku: 'ErGw1AZ'
+  vpnType: 'RouteBased'
+  vpnGatewayGeneration: 'None'
+  enableBgp: false
+  activeActive: false
+  enableBgpRouteTranslationForNat: false
+  enableDnsForwarding: false
+  bgpPeeringAddress: ''
+  bgpsettings: {
+    asn: '65515'
+    bgpPeeringAddress: ''
+    peerWeight: '5'
+  }
+}
+
+@sys.description('''Resource Lock Configuration for ExpressRoute Virtual Network Gateway.
+
+- `kind` - The lock settings of the service which can be CanNotDelete, ReadOnly, or None.
+- `notes` - Notes about this lock.
+
+''')
+param parVirtualNetworkGatewayLock lockType = {
+  kind: 'None'
+  notes: 'This lock was created by the ALZ Bicep Hub Networking Module.'
+}
+
+@sys.description('Tags you would like to be applied to all resources in this module.')
+param parTags object = {}
+
+@sys.description('Set Parameter to true to Opt-out of deployment telemetry.')
+param parTelemetryOptOut bool = false
+
+@sys.description('Define outbound destination ports or ranges for SSH or RDP that you want to access from Azure Bastion.')
+param parBastionOutboundSshRdpPorts array = [ '22', '3389' ]
+
+var varLocations = [
```

Comment:
If we will deploy just one region using this module (the secondary region), then I think we can just use the `parLocation` parameter